### PR TITLE
work on #427

### PR DESCRIPTION
**Pull request by fabianfrz**

### CODE REVIEW

File: src/www/diag_logs_filter.php [hunk 1]
```diff
@@ -745,7 +745,7 @@ function getGETPOSTsettingvalue($settingname, $default)
 									  <td class="listMRr nowrap"><?php echo htmlspecialchars($filterent['time']);?></td>
 									  <td class="listMRr nowrap">
 										<?php if ($filterent['direction'] == "out"): ?>
-										<span class="glyphicon glyphicon-cloud-download" alt="Direction=OUT" title="Direction=OUT"></span>
+                    <span class="glyphicon glyphicon-cloud-download" alt="<?= gettext('Direction=OUT') ?>" title="<?= gettext('Direction=OUT') ?>"></span>
```
Comment (fichtner) [inline src/www/diag_logs_filter.php]: This looks more like programming than something that is easy to read or to translate! Either we drop this translation or clean it up further?

File: src/www/diag_sockets.php [hunk 2]
```diff
@@ -49,9 +49,9 @@
 
 			<section class="col-xs-12">
 
-                <p>Information about listening sockets for both <a href="#IPv4">IPv4</a> and <a href="#IPv6">IPv6</a>.</p>
-                <p>For explanation about the meaning of the information listed for each socket click <a href="#about">here</a>.</p>
-                <p><input type="button" class="btn btn-default" value="<?=$showAllText?>" onclick="window.location.href='diag_sockets.php<?=$showAllOption?>'"/><br/>To show information about both listening and connected sockets click this.</p>
+      <p><?= sprintf(gettext('Information about listening sockets for both %s and %s.'), sprintf('<a href="#IPv4">%s</a>', gettext('IPv4')),sprintf('<a href="#IPv6">%s</a>', gettext('IPv6'))) ?></p>
```
Comment (fichtner) [inline src/www/diag_sockets.php]: Please don't nest sprintf, use %sIPv4%s and %sIPv6%s

File: src/www/diag_sockets.php [hunk 3]
```diff
@@ -120,21 +120,20 @@
...
+          <tr><td class="listlr">FD	      </td><td class="listr"><?= gettext('The file descriptor number of the socket.') ?></td></tr>
+          <tr><td class="listlr">PROTO	      </td><td class="listr"><?= gettext('The transport protocol associated with the socket for Internet sockets, or the type of socket (stream or data-gram) for UNIX sockets.') ?></td></tr>
+          <tr><td class="listlr">ADDRESS	      </td><td class="listr"><?= sprintf(gettext("(UNIX sockets only) For bound sockets, this is the file-name of the socket. %sFor other sockets, it is the name, PID and file descriptor number of the peer, or ``(none)'' if the socket is neither bound nor connected."),'<br />') ?></td></tr>
+          <tr><td class="listlr">LOCAL ADDRESS    </td><td class="listr"><?= gettext('(Internet sockets only) The address the local end of the socket is bound to (see getsockname(2)).') ?></td></tr>
+          <tr><td class="listlr">FOREIGN ADDRESS  </td><td class="listr"><?= gettext('(Internet sockets only) The address the foreign end of the socket is bound to (see getpeername(2)).') ?></td></tr>
```
Comment (fichtner) [inline src/www/diag_sockets.php]: Could we drop the superfluous  whitespace in the first column? :)

File: src/www/diag_system_pftop.php [hunk 4]
```diff
@@ -150,7 +150,7 @@ function getpftopactivity() {
                       <option value='200' selected="selected">200</option>
                       <option value='500'>500</option>
                       <option value='1000'>1000</option>
-                      <option value='9999999999'>all</option>
+                      <option value='9999999999'><?= gettext('all') ?></option>
```
Comment (fichtner) [inline src/www/diag_system_pftop.php]: Unrelated side note :D

    $infinity =  9999999999;
    echo "To $infinity and beyond.";

File: src/www/firewall_nat_out_edit.php [hunk 5]
```diff
@@ -447,10 +447,10 @@ function formTranslateAddresses() {
                   <td>
                     <div class="input-group">
                       <select id="proto" name="protocol" class="selectpicker" data-live-search="true" data-size="5" data-width="auto">
-<?php                foreach (explode(" ", "any TCP UDP TCP/UDP ICMP ESP AH GRE IPV6 IGMP PIM OSPF") as $proto):
+<?php                foreach (array("any" => gettext("any"), "TCP" => gettext("TCP"), "UDP" => gettext("UDP"), "TCP/UDP" => gettext("TCP/UDP"), "ICMP" => gettext("ICMP"), "ESP" => gettext("ESP"), "AH" => gettext("AH"), "GRE" => gettext("GRE"), "IPV6" => gettext("IPV6"), "IGMP" => gettext("IGMP"), "PIM" => gettext("PIM"), " OSPF" => gettext("OSPF")) as $proto => $proto_translated):
 ?>
               <option value="<?=strtolower($proto);?>" <?= strtolower($proto) == $pconfig['protocol'] ? "selected=\"selected\"" : ""; ?>>
-                          <?=$proto;?>
+                          <?=$proto_translated?>
```
Comment (fichtner) [inline src/www/firewall_nat_out_edit.php]: Please only a single line for <option ...>TEXT</option>

File: src/www/services_dhcp_edit.php [hunk 6]
```diff
@@ -397,16 +397,16 @@ function show_tftp_config() {
...
-									    <br /> <span class="vexpl">Name of the file that should be loaded when this host boots off of the network, overrides setting on main page.</span></td>
+                      <br /> <span class="vexpl"><?= gettext('Name of the file that should be loaded when this host boots off of the network, overrides setting on main page.') ?></span></td>
 									</tr>
 									<tr>
-									  <td width="22%" valign="top" class="vncell">Root Path</td>
+                    <td width="22%" valign="top" class="vncell"><?= gettext('Root Path') ?></td>
 									  <td width="78%" class="vtable">
 										<input name="rootpath" type="text" class="formfld unknown" id="rootpath" size="90" value="<?=htmlspecialchars($pconfig['rootpath']);?>" />
-									    <br /> <span class="vexpl"><?=gettext("Enter the"); ?> <b><?=gettext("root-path"); ?></b>-<?=gettext("string");?>, overrides setting on main page.</span></td>
+                      <br /> <span class="vexpl"><?=sprintf(gettext("Enter the %sroot-path%s-string, overrides setting on main page."), '<b>','</b>') ?></span></td>
```
Comment (fichtner) [inline src/www/services_dhcp_edit.php]: No need for bold tags.

File: src/www/services_dhcpv6_edit.php [hunk 7]
```diff
@@ -198,7 +198,7 @@ function staticmaps_sort($ifgui) {
 					                    <input name="duid" type="text" class="formfld unknown" id="duid" size="40" value="<?=htmlspecialchars($pconfig['duid']);?>" />
 					                    <br />
 					                    <span class="vexpl"><?=gettext("Enter a DUID Identifier in the following format: ");?><br />
-					"DUID-LLT - ETH -- TIME --- ---- address ----" <br />
+          "<?= gettext('DUID-LLT - ETH -- TIME --- ---- address ----') ?>" <br />
```
Comment (fichtner) [inline src/www/services_dhcpv6_edit.php]: Only adds to confusion, maybe change the string to "DUID-LLT - ETH -- TIME --- ---- ADDR ----" and leave it static?

File: src/www/services_dhcpv6_edit.php [hunk 8]
```diff
@@ -219,16 +219,16 @@ function staticmaps_sort($ifgui) {
...
-							    <br /> <span class="vexpl">Name of the file that should be loaded when this host boots off of the network, overrides setting on main page.</span></td>
+                  <br /> <span class="vexpl"><?= gettext('Name of the file that should be loaded when this host boots off of the network, overrides setting on main page.') ?></span></td>
 							</tr>
 							<tr>
-							  <td width="22%" valign="top" class="vncell">Root Path</td>
+                <td width="22%" valign="top" class="vncell"><?= gettext('Root Path') ?></td>
 							  <td width="78%" class="vtable">
 								<input name="rootpath" type="text" class="formfld unknown" id="rootpath" size="90" value="<?=htmlspecialchars($pconfig['rootpath']);?>" />
-							    <br /> <span class="vexpl"><?=gettext("Enter the"); ?> <b><?=gettext("root-path"); ?></b>-<?=gettext("string");?>, overrides setting on main page.</span></td>
+                  <br /> <span class="vexpl"><?= sprintf(gettext('Enter the %sroot-path%s-string, overrides setting on main page.'),'<b>','</b>') ?></span></td>
```
Comment (fichtner) [inline src/www/services_dhcpv6_edit.php]: no need for bold here too

**Comment by fichtner** [comment]
Ready for merge? I'm going to flatten into atomic commit...

**Comment by fabianfrz** [comment]
you can merge it but one change is not done

**Comment by fichtner** [comment]
Works for me. Let's bring this in, thanks. :)

**Comment by fichtner** [comment]
Trailing whitespace are mostly wrong (tabs vs. spaces), please take more care to leave them as is next time. Merged via https://github.com/opnsense/core/commit/6612086c2135ce8b13a93915822694d2a78b5225 and this is closed, cheers!

**Pull request closed without merging**